init.lua

```diff
@@ -1,77 +1,302 @@
+wielded_light = {}
+
+function wielded_light.register_lightstep(callback)
+	table.insert(update_callbacks, callback)
+end
+
+function wielded_light.register_player_lightstep(callback)
+	table.insert(update_player_callbacks, callback)
+end
+
+function wielded_light.wielded_light_of_level(light_level)
+	return mod_name..":"..light_level
+end
+
+function wielded_light.level_of_wielded_light(node_name)
+	return tonumber(node_name:sub(#mod_name+2))
+end
+
+function wielded_light.is_wielded_light(node_name)
+	return node_name:sub(1, #mod_name) == mod_name
+end
+
+function wielded_light.is_lightable_node(node_pos)
+	local name = minetest.get_node(node_pos).name
+	if name == "air" then
+		return true
+	elseif wielded_light.is_wielded_light(name) then
+		return true
+	end
+	return false
+end
+
+function wielded_light.get_light_position(pos)
+	local around_vector = {
+		{x=0, y=0, z=0},
+		{x=0, y=1, z=0}, {x=0, y=-1, z=0},
+		{x=1, y=0, z=0}, {x=-1, y=0, z=0},
+		{x=0, y=0, z=1}, {x=0, y=0, z=1},
+	}
+	for _, around in ipairs(around_vector) do
+		local light_pos = vector.add(pos, around)
+		if wielded_light.is_lightable_node(light_pos) then
+			return light_pos
+		end
+	end
+end
+
+function wielded_light.get_light_level(item_name)
+	if not item_name or item_name == "" then
+		return 0
+	end
+
+	local cached_light_level = shiny_items[item_name]
+	if cached_light_level then
+		return cached_light_level
+	end
+
+	local stack = ItemStack(item_name)
+	local itemdef = stack:get_definition()
+	if not itemdef then
+		return 0
+	end
+	return math.min(math.max((itemdef.light_source or 0) - level_delta, 0), max_light_level)
+	-- local light_level = math.min(math.max((itemdef.light_source or 0) - level_delta, 0), max_light_level)
+	-- -- shiny_items[item_name] = light_level
+
+	-- return light_level
 end
 
 function wielded_light.register_item_light(itemname, light_level)
 	shiny_items[itemname] = light_level
 end
 
+function wielded_light.track_item_entity(userdata, cat, item)
+	local light_level = wielded_light.get_light_level(item)
+	if light_level <= 0 then return end
+
+	local uid = tostring(userdata)
+	local id = get_cat_code(cat)..uid
+	if not tracked_entities[uid] then
+		local pos = entity_pos(userdata)
+		local pos_str = pos and minetest.pos_to_string(pos)
+		if pos_str then
+			add_light(pos_str, id, light_level)
+		end
+		tracked_entities[uid] = { obj=userdata, items={}, pos=pos_str, update = true }
+	end
+	tracked_entities[uid].items[id] = { level=light_level }
+end
+
+local player_height_offset = { x=0, y=1, z=0 }
+function wielded_light.track_user_entity(userdata, cat, item)
+	local uid = tostring(userdata)
+	local id = get_cat_code(cat)..uid
+	if not tracked_entities[uid] then
+		tracked_entities[uid] = { obj=userdata, items={}, offset = player_height_offset, update = true }
+	end
+	local tracked_entity = tracked_entities[uid]
+	local tracked_item = tracked_entity.items[id]
+
+	if not tracked_item
+	or tracked_item.item ~= item
+	then
```

Comment:
Weird indent level. A single line for all of them would be fine.

Reply:
It's easier for commenting out one part of the condition at a time, this one was probably lot bigger before it was simplified